ql/math/randomnumbers/fractionalnoisersg.hpp

```diff
@@ -0,0 +1,149 @@
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+/*! \file fractionalnoisersg.hpp
+    \brief Fractional Gaussian noise (sequence) generator
+*/
+
+#ifndef quantlib_fractional_gaussian_noise_rsg_hpp
+#define quantlib_fractional_gaussian_noise_rsg_hpp
+
+#include <ql/math/fastfouriertransform.hpp>
+#include <ql/math/matrix.hpp>
+#include <ql/methods/montecarlo/sample.hpp>
+#include <complex>
+
+using std::pow;
+
+namespace QuantLib {
+
+    //! Fractional Gaussian noise (sequence) generator
+    /*! Using the method by Davies & Harte, improved by Chan & Wood, of finding
+        the square root of the autocovariance matrix by means of embedding 
+        it in a circulant matrix.
+        -# Andrew T. A. Wood, & Chan, G. (1994). Simulation of Stationary 
+        Gaussian Processes in [0,1]^d. Journal of Computational and Graphical
+        Statistics, 3(4), 409–432. https://doi.org/10.2307/1390903
+
+        \test the correctness of the returned values is tested by 
+              a likelihood-ratio test (LRT) of covariance matrix 
+    */
+    template <class GSG>
+    class FractionalGaussianNoiseRsg {
+      public:
+        typedef Sample<std::vector<Real> > sample_type;
+        FractionalGaussianNoiseRsg(Real HurstParameter,
+                                   GSG generator);
+        const sample_type& nextSequence() const;
+        const sample_type& lastSequence() const;
+        Size dimension() const;
+        Real gamma(Size k) const { return k==0 ? 1.0 : 0.5 * (pow(k-1,2*H_)-2*pow(k,2*H_)+pow(k+1,2*H_)); };
+      private:
+        Real H_;
+        GSG generator_;
+        Size n_;
+        Size m_;
+        Size log2_m_;
+        FastFourierTransform fft_;
+        /// @brief Square roots of eigenvalues of circulant matrix
+        std::vector<Real> sqrt_lambda_;
+        mutable std::vector<std::complex<Real> > fft_in_;
+        mutable std::vector<std::complex<Real> > fft_out_;
+        mutable sample_type next_;
+        /// @brief Floor of log_2(n).
+        Size log_2(Size n) const { Size l=0; while(n>1) {n/=2;l++;} return l;}
+    };
+
+    // template definitions
+
+    template <class GSG>
+    FractionalGaussianNoiseRsg<GSG>::FractionalGaussianNoiseRsg(Real HurstParameter, GSG generator)
+    : H_(HurstParameter),
+      generator_(std::move(generator)), 
+      n_(generator_.dimension()),
+      m_(2*n_),
+      log2_m_(log_2(m_)),
+      fft_(log2_m_),
+      sqrt_lambda_(m_), 
+      fft_in_(m_), fft_out_(m_), 
+      next_(std::vector<Real> (n_), 1.0)
+    {
+        Size bin_floor_m_ = static_cast<std::size_t>(1) << log2_m_;
```

Comment:
You should probably use `static_cast<Size>(1)` for consistency. Also you can use `auto` on the left-hand side to avoid specifying the type more than once.